contracts/wrapper/NameWrapper.sol

```diff
@@ -127,9 +127,19 @@ contract NameWrapper is
     {
         (owner, fuses, expiry) = super.getData(id);
 
-        bytes32 labelhash = _getEthLabelhash(bytes32(id), fuses);
-        if (labelhash != bytes32(0)) {
-            expiry = uint64(registrar.nameExpires(uint256(labelhash))) + GRACE_PERIOD;
+        bytes32 labelHash = _getEthLabelhash(bytes32(id), fuses);
+        if (labelHash != bytes32(0)) {
+            uint64 registrarExpiry = uint64(
+                registrar.nameExpires(uint256(labelHash))
+            );
+            expiry = registrarExpiry + GRACE_PERIOD;
+            // if owner in registrar is not the wrapper, zero out the owner
+            if (
```

Comment:
Hm, does this mean that a name that's subject to the vulnerability will briefly appear to have an owner again during the grace period? We should probably zero out the owner if the expiry is in the past even if it *is* in grace.

Reply:
Isn't this already doing that?

Reply:
My mistake.